app/src/main/java/com/github/alfabravo2013/readyforexams/domain/login/LoginUseCase.kt

```diff
@@ -0,0 +1,26 @@
+package com.github.alfabravo2013.readyforexams.domain.login
+
+import com.github.alfabravo2013.readyforexams.R
+import com.github.alfabravo2013.readyforexams.presentation.login.LoginViewModel.OnEvent
+import com.github.alfabravo2013.readyforexams.util.Result
+import com.github.alfabravo2013.readyforexams.util.isInvalidEmail
+import com.github.alfabravo2013.readyforexams.util.isInvalidPassword
+
+class LoginUseCase(private val repository: LoginRepository) {
+
+    suspend fun login(email: String, password: String): OnEvent {
+        if (email.isInvalidEmail()) {
```

Comment:
No R.string values here. I would suggest to use plain strings "invalid email error text" . Ideally, values like this will come from server.
OnEvent class should not be imported here. You inject useCase inside the ViewModel but no values from the ViewModel here. 
No NavigateToHomeScreen event here. Most of the times here you only care about Success or Failure with error.